grading/clean.cpp

```diff
@@ -308,7 +313,23 @@ bool isNumber(const std::string &str) {
       return false;
     }
   }
-  return true == atLeastOneDigit;
+  return atLeastOneDigit;
+}
+
+std::string isolateAlphanumAndNumberPunctuation(const std::string &str) {
```

Comment:
this function is somewhat confusing in purpose and should at least have a comment at the start...